src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public class Ball {
+    private ColorSupplier.Color color;
+    private int number;
+
+    public Ball(ColorSupplier.Color color, int number) {
+        this.color = color;
+        this.number = number;
+    }
+    //getter and setter to color
+    public ColorSupplier.Color getColor() {
+        return color;
+    }
+
+    public void setColor() {
+        this.color = color;
+    }
+    //getter and setter to number
+    public int getNumber() {
+        return number;
+    }
+
+    public void setNumber() {
+        this.number = number;
+    }
+
+    public String toString () {
+        return "Ball{color='" + color + "', number=" + number + "}";
+    }
+}
```

Comment:
The class and its methods should have access modifiers. In this case, the class should be 'public', and the methods should have appropriate access modifiers based on their intended visibility.